### DISCUSSION TITLE
Dockerize services for local dev work

### DISCUSSION THREAD

**Discussion by codecakes** [Django-CRM/Django-CRM#524] [Ideas]
TL;DR: #523 

Objective: Make it easy for local crm development to be as decoupled from dependent services.

Explanation: As in PR #523 proposal, I found few glitches to setting up and running the api that could require workaround if doing a full swing development. If we can end up making changes and just testing it by running `docker compose -f docker/docker-compose.yml up` with some opinionated services for queue brokering and backend storage, then driving the development forward would be easier.

Advantages:
- Just like how dependant services would be spun up on cloud such as redis, celery, postgres
- dependent services once containerized reduce time wasted on local metal configuration. we don't need to manually run, clear or flush redis, celery instances.
